### PR TITLE
Adopted use of new ERFA ecliptic transformation machinery

### LINKED ISSUE #4873: Use new ERFA ecliptic transformation machinery?
As liberfa/erfa#34 points out there's a new SOFA release, which means we'll soon have a new ERFA release to include in astropy.

This new release includes a new pair of functions to go from equatorial to ecliptic and back again. Probably we will want to use this as a new implementation for the ecliptic transformations? An open question is whether we want to just use these in place of the current machinery or if there's some subtle difference in implementation.  Will have to see how it goes once a new liberfa is out and packaged up.

If it turns out this gives substantially different answers we may want to consider trying to get this in for 1.2.  Won't have time to do that by Friday, though... @astrofrog, do you agree this is worth exploring post feature-freeze? (given that it may actually be fixing a "bug" in some sense of the word in the ecliptic transformation)

cc @mhvk

### PR DESCRIPTION
**Pull request by homeboy445**
### Description

Adopted the use of new ERFA ecliptic transformation machinery in ecliptic_transforms.py

Fixes #4873

### CODE REVIEW

**Comment by pllim** [comment]
Thanks, will need a test and a change log, **but** before you proceed, let's wait for a review first, as the issue you are trying to fix is very old so I am not sure if it is still an issue or not...

File: astropy/coordinates/builtin_frames/ecliptic_transforms.py [hunk 1]
```diff
@@ -31,9 +31,8 @@ def _mean_ecliptic_rotation_matrix(equinox):
     # function and also brings the results closer to what other libraries
```
Comment (mhvk) [inline astropy/coordinates/builtin_frames/ecliptic_transforms.py:31]: I think with the change we also need to adjust the comment, explaining why this function exists at all! Maybe
```
This code just calls ecm06, which uses the precession matrix according to the
IAU 2006 model, but leaves out nutation. This brings the results closer to what
other libraries give (see https://github.com/astropy/astropy/pull/6508).
```

File: astropy/coordinates/builtin_frames/ecliptic_transforms.py [hunk 2]
```diff
@@ -31,9 +31,8 @@ def _mean_ecliptic_rotation_matrix(equinox):
...
     # give (see https://github.com/astropy/astropy/pull/6508).
     jd1, jd2 = get_jd12(equinox, 'tt')
-    rbp = erfa.pmat06(jd1, jd2)
```
Comment (mhvk) [inline astropy/coordinates/builtin_frames/ecliptic_transforms.py:34]: I would make this just a one-liner with
```
return erfa.ecm06(*get_jd12(equinix, 'tt'))
```

**Review by mhvk — CHANGES_REQUESTED** [review]
@homeboy445 - thanks for applying the fix. Since @astrojuanlu basically said this was what was needed to be done, no problems there!  However, the code can be even simpler, and, more importantly, could you change the comment above the code, otherwise it will just confuse people! Suggestion inline.

**Comment by pllim** [comment]
Thanks for your continued patience, @homeboy445 ! Can you please squash all your commits into one commit, given that there is a merge commit in there anyway?

**Review by astrojuanlu — APPROVED** [review]
Looks good to me!

**Review by mhvk — APPROVED** [review]
Ah, I hadn't noticed the change was already made. All OK now. Thanks, @homeboy445!

**Pull request merged**